documents/FAQ.md

```diff
@@ -49,6 +49,9 @@ This could be many different things but it most likely is a policy issue. Start
 
 After getting it working make sure to only allow the actions and resources that you need. More info about IoT IAM policies can be found [here](https://docs.aws.amazon.com/iot/latest/developerguide/security_iam_service-with-iam.html).
 
+### I am experiencing deadlocks
+You MUST NOT perform blocking operations on any callback, or you will cause a deadlock. For example: in the on_publish_received callback, do not send a publish, and then wait for the future to complete within the callback. The Client cannot do work until your callback returns, so the thread will be stuck.
```

Comment:
Trivial: Some markdown editors show warnings if there is no empty line after the heading